vector/src/main/java/im/vector/app/features/settings/VectorSettingsPinFragment.kt

```diff
@@ -66,20 +66,24 @@ class VectorSettingsPinFragment @Inject constructor(
                 biometricHelper.isSystemKeyValid
     }
 
+    private fun shouldEnableBiometricPref(isPinCodeChecked: Boolean): Boolean {
+        return isPinCodeChecked && biometricHelper.canUseAnySystemAuth
+    }
+
     override fun onResume() {
         super.onResume()
 
-        useBiometricPref.isEnabled = usePinCodePref.isChecked
-        useBiometricPref.isChecked = shouldCheckBiometricPref(usePinCodePref.isChecked)
+        useBiometricPref.isEnabled = shouldEnableBiometricPref(isPinCodeChecked = usePinCodePref.isChecked)
```

Comment:
do you think it's worth extracting the `useBiometricPref.isEnabled  and useBiometricPref.isChecked` setting to a reusable function?